colossalai/communication/collective.py

```diff
@@ -8,10 +8,13 @@
 
 from colossalai.context import ParallelMode
 from colossalai.core import global_context as gpc
-from colossalai.utils import get_current_device
 
 
-def all_gather(tensor: Tensor, dim: int, parallel_mode: ParallelMode, async_op: bool = False) -> Tensor:
+def all_gather(tensor: Tensor,
+               dim: int,
+               parallel_mode: ParallelMode,
+               on_cpu: bool = False,
```

Comment:
on_cpu may have a conflict with tensor.device.
we have to ensure tensor.device is CPU?

Reply:
Let me fix. I think maybe we do not need this option, but just check if the tensor is on cpu.